Bridge rebased to latest from main

dmetzgar opened the pull request:
Includes Stephen's duplex tests, copyright info, and appdomain fixes.

dnfclas commented:
Hi __@dmetzgar__, I'm your friendly neighborhood .NET Foundation Pull Request Bot (You can call me DNFBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by .NET Foundation and real humans are currently evaluating your PR.</p>

TTYL, DNFBOT;

dmetzgar commented:
Rebased to main again and squashed all the changes, including the removal of Newtonsoft.Json from all the outerloop tests and the dependency on VS2015 for building the bridge.

SajayAntony commented on tools/Bridge/Bridge/App.config:
```diff
@@ -0,0 +1,26 @@
+  <runtime>
+    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
+      <dependentAssembly>
+        <assemblyIdentity name="Newtonsoft.Json" publicKeyToken="30ad4fe6b2a6aeed" culture="neutral" />
```
Can remove I think.

SajayAntony commented on tools/Bridge/Bridge/Bridge.csproj:
```diff
@@ -0,0 +1,119 @@
+    </Reference>
+    <Reference Include="Newtonsoft.Json, Version=6.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
+      <HintPath>$(PackageOutputDir)\Newtonsoft.Json.6.0.6\lib\net45\Newtonsoft.Json.dll</HintPath>
+      <Private>True</Private>
```
Do you still need this?

SajayAntony commented:
Once you get the buddy build done by Ron I think we should merge it :shipit:
